### PR TITLE
Game Configuration Register pull request

### PR DESCRIPTION
**Pull request by xKoem**
Created GameConfigurationRegister microservice with tests.

### CODE REVIEW

**Comment by boczkas** [comment]
Build 136 Still Failing

File: GameConfigurationRegister/pom.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    <parent>
+        <artifactId>battleSheep</artifactId>
+        <groupId>pl.jakubowskiprzemyslaw.tajgertim</groupId>
+        <version>1.0</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>GameConfigurationRegister</artifactId>
+
+
```
Comment (KrzysztofDzioba) [inline GameConfigurationRegister/pom.xml]: enter

File: GameConfigurationRegister/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationQueueListener.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import pl.jakubowskiprzemyslaw.tajgertim.models.configuration.GameConfiguration;
+import pl.jakubowskiprzemyslaw.tajgertim.services.LoggerService;
+
+@Service
+class GameConfigurationQueueListener {
+
+    private final LoggerService loggerService;
+    private final ApplicationEventPublisher publisher;
+
+    @Autowired
```
Comment (KrzysztofDzioba) [inline GameConfigurationRegister/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationQueueListener.java]: is it necessary?

https://docs.spring.io/spring/docs/4.3.x/spring-framework-reference/htmlsingle/#beans-autowired-annotation

File: GameConfigurationRegister/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfiguredEventListener.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        GameConfiguredEventListener that = (GameConfiguredEventListener) o;
+        return Objects.equals(queueService, that.queueService);
+    }
+
+    @Override
+    public int hashCode() {
+
```
Comment (KrzysztofDzioba) [inline GameConfigurationRegister/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfiguredEventListener.java]: enter

File: GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationEventTest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    @BeforeMethod
+    public void setUp() {
+        GameConfigurationQueueListener queueListener = mock(GameConfigurationQueueListener.class);
+        configuration = new GameConfiguration();
+        event = new GameConfiguredEvent(queueListener, configuration);
+    }
+
+    public void equalsTheSame_ReturnsTrue() {
+        assertEquals(event, event);
```
Comment (KrzysztofDzioba) [inline GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationEventTest.java]: really? ;d

File: GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationEventTest.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    public void equalsNull_ReturnsFalse() {
+        assertNotEquals(event, null);
+    }
+
+    public void equalsAnotherObject_ReturnsTrue_IfAnotherObjectIsSimilar() {
+        GameConfiguredEvent configuredEvent = new GameConfiguredEvent(mock(GameConfigurationQueueListener.class), configuration);
+        assertEquals(event, configuredEvent);
```
Comment (KrzysztofDzioba) [inline GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationEventTest.java]: what does it check?

File: GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfiguredEventListenerTest.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import pl.jakubowskiprzemyslaw.tajgertim.services.QueueService;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+@Test
+public class GameConfiguredEventListenerTest {
+
+
+    public void correctObjectInGameConfiguration_SendsToQueueGameConfiguration_AndCallsEventListener() {
```
Comment (KrzysztofDzioba) [inline GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfiguredEventListenerTest.java]: //given
//when
//then

File: GameConfigurationRegister/pom.xml (hunk 1 above)
Comment (boczkas) [reply GameConfigurationRegister/pom.xml]: done

File: GameConfigurationRegister/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationQueueListener.java (hunk 2 above)
Comment (boczkas) [reply GameConfigurationRegister/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationQueueListener.java]: done

File: GameConfigurationRegister/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfiguredEventListener.java (hunk 3 above)
Comment (boczkas) [reply GameConfigurationRegister/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfiguredEventListener.java]: done

File: GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationEventTest.java (hunk 4 above)
Comment (boczkas) [reply GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationEventTest.java]: done

File: GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationEventTest.java (hunk 5 above)
Comment (boczkas) [reply GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationEventTest.java]: done

File: GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfiguredEventListenerTest.java (hunk 6 above)
Comment (boczkas) [reply GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfiguredEventListenerTest.java]: done

File: GameConfigurationRegister/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfiguredEventListener.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    @Autowired
+    public GameConfiguredEventListener(QueueService queueService) {
+        this.queueService = queueService;
+    }
+
+    @Override
+    public void onApplicationEvent(GameConfiguredEvent event) {
+        GameConfiguration gameConfiguration = event.getGameConfiguration();
+        queueService.sendObjectToQueue(Queues._3FleetPlacementSizeQueue, gameConfiguration);
```
Comment (RafalJDev) [inline GameConfigurationRegister/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfiguredEventListener.java]: /n for clarity

File: GameConfigurationRegister/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameRegisterConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+import pl.jakubowskiprzemyslaw.tajgertim.services.LoggerService;
+import pl.jakubowskiprzemyslaw.tajgertim.services.QueueService;
+import pl.jakubowskiprzemyslaw.tajgertim.services.SessionService;
+
+@Configuration
+class GameRegisterConfiguration {
+
+    private final RabbitTemplate rabbitTemplate;
+
+    @Autowired
```
Comment (RafalJDev) [inline GameConfigurationRegister/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameRegisterConfiguration.java]: @Autowired is redundant

Comment (JaroslawSlaby) [reply GameConfigurationRegister/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameRegisterConfiguration.java]: done

File: GameConfigurationRegister/src/main/resources/application.properties [hunk 9]
```diff
@@ -0,0 +1,5 @@
+spring.rabbitmq.addresses=10.30.1.140
+spring.rabbitmq.username=admin
```
[the remark this answers is not in the document]
Comment (JaroslawSlaby) [reply GameConfigurationRegister/src/main/resources/application.properties]: yeah, I know

File: GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationQueueListenerTest.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+public class GameConfigurationQueueListenerTest {
+
+    private GameConfigurationQueueListener gameConfigurationQueueListener;
+    private ApplicationEventPublisher publisher;
+    private GameConfiguration gameConfiguration;
+
+    @BeforeMethod
+    public void setUp() {
+        gameConfiguration = new GameConfiguration();
+        publisher = mock(ApplicationEventPublisher.class);
```
[the remark this answers is not in the document]
Comment (JaroslawSlaby) [reply GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationQueueListenerTest.java]: done

File: GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationQueueListenerTest.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    public void correctObjectInGameConfiguration_SendsToQueueGameConfiguration() {
+        // given
+        gameConfigurationQueueListener.getGameConfigurationFromGameConfigurationQueue(gameConfiguration);
+
+        // when
+        GameConfiguredEvent event = new GameConfiguredEvent(this, gameConfiguration);
+
+        // then
```
[the remark this answers is not in the document]
Comment (JaroslawSlaby) [reply GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfigurationQueueListenerTest.java]: thanks

File: GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfiguredEventListenerTest.java [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        // given
+        QueueService queueService = mock(QueueService.class);
+        GameConfiguredEventListener gameConfigurationEventListener = new GameConfiguredEventListener(queueService);
```
[the remark this answers is not in the document]
Comment (JaroslawSlaby) [reply GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfiguredEventListenerTest.java]: done

File: GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfiguredEventListenerTest.java [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+import pl.jakubowskiprzemyslaw.tajgertim.queues.Queues;
+import pl.jakubowskiprzemyslaw.tajgertim.services.QueueService;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+@Test
+public class GameConfiguredEventListenerTest {
+
+
```
[the remark this answers is not in the document]
Comment (boczkas) [reply GameConfigurationRegister/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/gameconfigurationregister/sendingconfig/GameConfiguredEventListenerTest.java]: done

**Pull request merged**